pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/PrestoDateTruncFunction.java

```diff
@@ -0,0 +1,262 @@
+      super("quarterOfYear");
+    }
+
+    @Override
+    public DurationFieldType getDurationType() {
+      return QUARTER_OF_YEAR_DURATION_FIELD_TYPE;
+    }
+
+    @Override
+    public DurationFieldType getRangeDurationType() {
+      return DurationFieldType.years();
+    }
+
+    @Override
+    public DateTimeField getField(Chronology chronology) {
+      return new OffsetDateTimeField(new DividedDateTimeField(new OffsetDateTimeField(chronology.monthOfYear(), -1), QUARTER_OF_YEAR, 3), 1);
+    }
+
+    private static class QuarterOfYearDurationFieldType
+            extends DurationFieldType {
+      private static final long serialVersionUID = -8167713675442491871L;
+
+      public QuarterOfYearDurationFieldType() {
+        super("quarters");
+      }
+
+      @Override
+      public DurationField getField(Chronology chronology) {
+        return new ScaledDurationField(chronology.months(), QUARTER_OF_YEAR_DURATION_FIELD_TYPE, 3);
+      }
+    }
+  }
+
+  public static final DateTimeFieldType QUARTER_OF_YEAR = new QuarterOfYearDateTimeField();
+
+  @Override
+  public String getName() {
+    return FUNCTION_NAME;
+  }
+
+  private static DateTimeField getTimestampField(ISOChronology chronology, String unitString) {
+    switch (unitString) {
+      case "millisecond":
+        return chronology.millisOfSecond();
+      case "second":
+        return chronology.secondOfMinute();
+      case "minute":
+        return chronology.minuteOfHour();
+      case "hour":
+        return chronology.hourOfDay();
+      case "day":
+        return chronology.dayOfMonth();
+      case "week":
+        return chronology.weekOfWeekyear();
+      case "month":
+        return chronology.monthOfYear();
+      case "quarter":
+        return QUARTER_OF_YEAR.getField(chronology);
+      case "year":
+        return chronology.year();
+    }
+    throw new IllegalArgumentException("'" + unitString + "' is not a valid Timestamp field");
+  }
+
+  // Usage:
+  // prestoDateTrunc(input, 'tz', 'unit') -> output.
+  // input and output are both long milliseconds since utc epoch
+  @Override
+  public void init(@Nonnull List<TransformFunction> arguments, @Nonnull Map<String, DataSource> dataSourceMap) {
+    if (arguments.size() != 4) {
+      throw new IllegalArgumentException(String.format("Exactly 4 arguments are required for %s transform function", FUNCTION_NAME));
+    }
+
+    TransformFunction firstArgument = arguments.get(0);
+    if (firstArgument instanceof LiteralTransformFunction || !firstArgument.getResultMetadata().isSingleValue()) {
+      throw new IllegalArgumentException(
+              "The first argument of prestoDateTrunc transform function must be a single-valued column or a transform function");
+    }
+    _mainTransformFunction = firstArgument;
+    _resultMetadata = LONG_SV_NO_DICTIONARY_METADATA;
+    String inputTimeUnitStr = ((LiteralTransformFunction) arguments.get(1)).getLiteral().toUpperCase(ENGLISH);
```

Comment:
(nit) Use `toUpperCase()` (default locale) for readability and consistency. Same for line 238 and 240